### PR TITLE
Describe local test run, fix some mage issues

### LINKED ISSUE #658: Describe local test run
There are no descriptions or prepared scripts (like make-commands) for test run. Setting up test environment requires using mage -- it is not a default `go mod tidy; go test .` run and not obvious for one unfamiliar with mage. Running a sequence of commands from CI was not enough for me: it failed with
```bash
Running unit tests...
../generate_templates_vfs.go:9:2: no required module provides package github.com/shurcooL/vfsgen; to add it:
	go get github.com/shurcooL/vfsgen
cli/codegen/static/static.go:5: running "go": exit status 1
Error: running "go generate -tags=dev ./cli/codegen/static" failed with exit code 1
```

### PR DESCRIPTION
**Pull request by DifferentialOrange**
To solve #658, I studied how current mage/CI pipeline of this repo works and met some issues, which was also fixed in this PR.

After PR #464 calling `mage clean` was broken. After execution of `mage generateGoCode` FSFile was created in path `cli/create/create_vfsdata_gen.go`, while `mage clean` tried to remove it from `cli/codegen/static/cartridge_vfsdata_gen.go` path.

Before this patch, calling `mage integration`, `mage testExamples` or `mage e2e` failed if `mage GenerateGoCode` hasn't been called yet (explicitly or as `mage lint`, `mage unit` or `mage build` dependency). Now all targets that require generated code explicitly depends on it.

[Documentation](https://magefile.org/dependencies/) says "Dependencies are guaranteed to run exactly once in a single execution of mage", but if there are several mage executions (like in this repo CI), dependencies will run multiple times. After this patch, Go code will not be rebuilt with `GenerateGoCode()` if build artifacts present, so CI will not do multiple deps calls. If you want to rebuild the code, call `mage clean` first.

At last, this path adds README.dev.md file with list of test requirements and short mage test commands descriptions. Scenario was validated in docker container.

Base image: `docker run -it --privileged cruizba/ubuntu-dind`

Commands:
```bash
export DEBIAN_FRONTEND=noninteractive
apt-get update

apt-get install -y wget
wget https://go.dev/dl/go1.17.5.linux-amd64.tar.gz
rm -rf /usr/local/go && tar -C /usr/local -xzf go1.17.5.linux-amd64.tar.gz
export PATH=$PATH:/usr/local/go/bin

apt-get install -y python3 pip
apt-get install -y unzip rpm cpio
apt-get install -y git
# docker already installed in dind

apt-get install -y curl
curl -L https://tarantool.io/OtKysgx/release/2.8/installer.sh | bash
apt-get install -y tarantool

git clone https://github.com/magefile/mage
cd mage
go run bootstrap.go
export PATH=$(go env GOPATH)/bin:$PATH

pip3 install pytest

git clone git@github.com:tarantool/cartridge-cli.git
cd ./cartridge-cli

git config --global user.email "test@tarantool.io"
git config --global user.name "Tar Antool"

pip3 install -r test/requirements.txt

tarantoolctl rocks install luacheck

mage test
```

I didn't forget about

- Tests (CI not fails, manually validated the guide)
- [x] Changelog
- [x] Documentation

Closes #658

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+## How to run tests
```
Comment (Totktonada) [inline README.dev.md:1]: FYI: I usually call it HACKING.md, however I don't whether it is widely spread.

Feel free to accept it if you like it or ignore.

**Review by Totktonada — APPROVED** [review]
I didn't validated the changes, but all of them looks meaningful. Thanks!

**Review by LeonidVas — COMMENTED** [review]
"Skip code generation if files exist"

**Comment by DifferentialOrange** [comment]
> Hi! Thank you for the patchset. "Skip code generation if files exist" - honestly, I would prefer generate the same files several times instead of adding the need to call "clean", because during development you can forget to call "clean" and think for a long time what went wrong. How badly does it hurt us to repeatedly generate these files during CI?

Well, the process seems quick by itself, but it would be called 5 times.

File: README.dev.md (hunk 1 above)
Comment (DifferentialOrange) [reply README.dev.md:1]: I've seen several README.dev.md examples, so I think it's legal.

**Comment by DifferentialOrange** [comment]
> Hi! Thank you for the patchset. "Skip code generation if files exist" - honestly, I would prefer generate the same files several times instead of adding the need to call "clean", because during development you can forget to call "clean" and think for a long time what went wrong. How badly does it hurt us to repeatedly generate these files during CI?

Ok, I reverted that one. @Totktonada , do you re-approve?

File: README.dev.md (hunk 1 above)
Comment (Totktonada) [reply README.dev.md:1]: Okay.

**Review by Totktonada — APPROVED** [review]
I'm ok.

**Review by LeonidVas — APPROVED** [review]

**Pull request merged**